### ISSUE TITLE
Windows Spotlight "Learn more about this picture" desktop icon issue

### ISSUE THREAD

**Issue by abdullah-erturk** [moudey/Shell#354]
After installing Nilesoft, Windows Spotlight Learn more about this picture when you right-click on the desktop icon, the options do not appear.

Windows 11 version: 22H2 v22621.2428

**Comment by crcastilho** [comment]
Same thing here.

**Comment by RubicBG** [comment]
can any of you take a screenshot before and after installing NS so I can get a better idea of the unwanted result

**Comment by crcastilho** [comment]
Right click on 
![image](https://github.com/moudey/Shell/assets/10247601/f222f3d0-3a91-4cab-9d70-42c41647b56a)
before NS gets this 
![image](https://github.com/moudey/Shell/assets/10247601/96f107c7-164e-4038-b2f1-6c2735316849)
and after gets this
![image](https://github.com/moudey/Shell/assets/10247601/78fe6a67-2f33-43f3-9c5a-d5f977425cfc)
Hope this helps.

**Comment by RubicBG** [comment]
It helps me understand the situation:

As you know, Microsoft has started to create a "modern" context menu (it has "Show more options" to display the "old" context menu). (I guess) Microsoft is trying with the modern context menu to limit the placement of commands by registry editing or by registering .dll files (restrict or control is still not clear) If you want to understand how commands are placed in the modern context menu you can watch this: https://www.youtube.com/watch?v=LfBQm-FY-2I

NS can NOT change the modern context menu - it "dresses" like a skin the old context menu with new options. When installing the program, it disables мodern context menu so it can run.

'Windows Spotlight Learn':
![Screenshot 2023-10-22 064032](https://github.com/moudey/Shell/assets/127151533/a537957a-1b89-4464-939e-fde0b9f5d7e7)![Screenshot 2023-10-22 063805](https://github.com/moudey/Shell/assets/127151533/b90fdc82-9208-46b2-82ad-173007216b6b)
I'm using it for the first time, but this is what the two menus look like on my computer (Win11 Pro 64-bit: 22621.2428)

I didn't know if it was NS script related - that's why I asked. Currently there is no such option to disable NS only for a given menu to show the modern context menu. I don't know if it's still possible because it might hinder his work

**Comment by abdullah-erturk** [comment]
The problem still persists.

**Comment by Rus1anB** [comment]
Unfortunately, the problem persists. I have no "Switch to next picture" option in the context menu from Shell :(

Canary version 1.9.14, Windows 11 Pro 23H2 22631.3085

**Comment by RubicBG** [comment]
"Switch to next picture" - and I don't have it anymore, I tried to restore it but without success - it is possible that this option was removed by Microsoft with some update

I need help, or rather information to see if there is anything I can do:
- I need a (older) version of windows where it is still working
- i need export for the registry key: HKEY_CLASSES_ROOT\CLSID\{2cc5ca98-6485-489a-920e-b3e88a6ccce3}\shell
There is the information about the commands, at the moment I only have SpotlightClick - SpotlightNext, SpotlightLike, SpotlightDislike are gone

**Comment by RubicBG** [comment]
![image](https://github.com/moudey/Shell/assets/127151533/6d823132-4a6a-4cb9-8e17-48d8289d3f21)
https://github.com/moudey/Shell/assets/127151533/d5385da3-b396-4205-92d4-611c34eb34b8
the first tests seem to be successful

**Comment by Rus1anB** [comment]
@RubicBG Cool, thanks!
Is this already coming in the new stable version or is it already in Canary?

**Comment by RubicBG** [comment]
- since the code depends entirely on what and how Microsoft feeds the information, any change by Microsoft the code may stop working - the code will be available as "Snippets" and will not be in a stable nor canary version
- set of 4 wallpapers are changed about a week apart - I'll need a week or two to check everything is working properly before I share the code
- if anyone wants to discuss something more on the topic, please visit https://discord.com/channels/1106387012707168318/

**Comment by RubicBG** [comment]
Windows Spotlight (beta)
https://nilesoft.org/forum/viewtopic.php?t=21
need feedback

**Comment by Rus1anB** [comment]
> need feedback

It's not working for me, see the video for details:

https://github.com/moudey/Shell/assets/153515121/d39c6c84-c318-43a0-b234-083a5768cfba

**Comment by RubicBG** [comment]
can play the video, pls resend

**Comment by Rus1anB** [comment]
Unfortunately, I've already removed this video from my PC((( But I have it playing on this page with no problems.
Try refreshing the page, or opening it in another browser, or saving it to your PC and watching it afterwards.

Of course I can download it from here and upload it again, but what will change?

If it still doesn't play for you, I'll re-record it tomorrow....

**Comment by RubicBG** [comment]
I was able to watch the video - I have an idea how to fix it, but I need time

**Comment by RubicBG** [comment]
NS has no function to read JSON syntax - I have to improvise. This time I have reduced the possibility of errors - but the more I limit it the more the code becomes "heavy" and the verification time increases
[Spotlight3.nss.txt](https://github.com/moudey/Shell/files/14522947/Spotlight3.nss.txt)
test this file (remove the .txt extension)
regardless if the code works or not i would like info if it is windows 11 or 10 and also the build version

**Comment by Rus1anB** [comment]
@RubicBG Cool, now it's working! Thanks!

![2024-03-08 124648](https://github.com/moudey/Shell/assets/153515121/40d3399d-9f96-4ac2-ae6e-1f5f6898d2d0)

Just please tell me how I can remove all unnecessary menu items from the Spotlight context menu, namely all those above the location name?

I apologize that the screenshot is my translation of the menu into Ukrainian, but there are items "Terminal", "File Manage" and "GoTo" at the top.

By the way, I need to remove these same menu items from the context menu of the Recycle Bin.

**Comment by Rus1anB** [comment]
Cool, it worked! 

It would be even cooler if in future versions you could actually change all of this in just one .nss file, so that you don't have to edit all the added menus separately each time you update.

_Already not entirely on the topic of the topic, but it would be great if, with some of the future versions of the program, the translation of all added items and sections of the context menu could be done in one single file! For example, I made a 100% translation of all Shell items into Ukrainian, but every time I update the version of your program, I have to MANUALLY check each .nss file for changes and manually update their translation, which is very inconvenient and takes a lot of time._

_If the entire translation can be done through just one single file, then this will make your program more popular among non-English speaking audiences! For example, I will be happy to share the Ukrainian translation for your program if there is such an opportunity in the future._

_Прошу прощения за офтоп._

**Comment by Rus1anB** [comment]
There is another problem with this context menu, which you can see in the screenshot. In the submenu of each wallpaper there are several empty menu items, clicking on which does not lead to anything:

![2024-03-08 173329](https://github.com/moudey/Shell/assets/153515121/f5a49f3d-268a-4eb6-acbc-ac849cd147a3)

**Comment by RubicBG** [comment]
`item(title='Spotlight JSON' cmd=io.file.write(path.combine(user.desktop, this.title +'.txt'), sys.ver.name + "\n\n" + sys.ver + "\n\n" + reg.get('HKCU\Software\Microsoft\Windows\CurrentVersion\DesktopSpotlight\Creatives', 'Creatives')))`

this command will generate a file on the desktop - send me the file. The structure in non-English sources may be different

**Comment by RubicBG** [comment]
> If the entire translation can be done through just one single file, ...

it is not a good idea to translate all menus and items with a single nss file. For each name, tool-tip, description, information in panels ..., a variable with a different name must be created. I have more than 100 nss files and some of them have more than 300 commands (like "Nilesoft Shell Theme Manager" https://discord.com/channels/1106387012707168318/1139275510506082336 with over 8600 item names and over 600 menu names)

what can be done is to split each nss file into two separate files - one will contain the code and the other only the translations. even if the code file changes, it can still recognize the translations from the second file. and the variable names will only apply to the first file

I am Bulgarian, but I prefer my menus to be in English

**Comment by RubicBG** [comment]
> Unfortunately, I do not get to the server using this link and cannot view the content

https://github.com/moudey/Shell/discussions/423#discussioncomment-8660246

**Comment by RubicBG** [comment]
[spotlight3.nss.txt](https://github.com/moudey/Shell/files/14542127/spotlight3.nss.txt)
Тhere is no information about these links in the code you sent me, that's why they don't appear for you. if the code contains this information with any of the following sessions of Windows Spotlight, they will appear

**Comment by Rus1anB** [comment]
> spotlight3.nss.txt

Awesome, now it's perfect! Thank you very much for your work!

**Comment by RubicBG** [comment]
https://nilesoft.org/forum/viewtopic.php?t=23

**Comment by Rus1anB** [comment]
> need feedback
> Is everything understandable?

Yes, from the description, I understand why it's needed and how it works.

Haven't tested it myself yet, because I'm too lazy to edit back all the .nss I've already fixed using your instructions above:

> where=not(wnd.is_desktop and sel.type==0) and ( the_existed_where_here )
